### PR TITLE
%pyproject_install macro should include --no-compile.

### PR DESCRIPTION
**Pull request by mcepl**
Fixes bsc#1094323

CC: @bmwiedemann

### CODE REVIEW

**Comment by bmwiedemann** [comment]
Yes, this should fix the problem of unreproducible .pyc files.
However, it also left the package I tested without .pyc files, which could give some percent penalty in startup time.

**Comment by bnavigator** [comment]
Maybe add something like the following to the `%pyproject_install` macro instead?

```spec
for d in %{buildroot}%{$python_sitelib} %{buildroot}%{$python_sitearch}; do
  if [ -d $d ]
    $python -m compileall $d
    $python -O -m compileall $d
  fi
done
```

File: macros/010-common-defs [hunk 1]
```diff
@@ -58,4 +58,6 @@
 %pyproject_wheel(+abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_-=) %{lua: local args = rpm.expand("%**") print(rpm.expand("%{python_expand $python -mpip wheel --no-deps %{?py_setup_args:--build-option %{py_setup_args}} --use-pep517 --no-build-isolation --progress-bar off --verbose . " .. args .. "}")) }
 
 # No such option: --strip-file-prefix %%{buildroot} 
-%pyproject_install(+abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_-=) %{lua: local args = rpm.expand("%**") print(rpm.expand("%{python_expand $python -mpip install --root %{buildroot}  --no-deps  --progress-bar off *.whl " .. args .. "}")) }
+%pyproject_install(+abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_-=) %{lua: local args = rpm.expand("%**") print(rpm.expand("%{python_expand $python -mpip install --root %{buildroot} --no-compile --no-deps  --progress-bar off *.whl " .. args .. "}")) }
+
+#vi:tw=0
```
Comment (mcepl) [inline macros/010-common-defs:63]: Please, don’t add modelines. Yes, most of use something vi-compatible, but do not make it a rule.

Comment (bnavigator) [reply macros/010-common-defs:63]: This is your own PR.

Comment (mcepl) [reply macros/010-common-defs:63]: Damn! Busted!

**Pull request merged**